drivers/SmartThings/zwave-thermostat/src/ct100-thermostat/init.lua

```diff
@@ -108,13 +113,31 @@ local function thermostat_mode_report_handler(self, device, cmd)
   if mode == ThermostatMode.mode.COOL or
     ((mode == ThermostatMode.mode.AUTO or mode == ThermostatMode.mode.OFF) and (current_temperature > (heating_setpoint + cooling_setpoint) / 2)) then
     device:send(ThermostatSetpoint:Get({setpoint_type = ThermostatSetpoint.setpoint_type.COOLING_1}))
-    device:send(ThermostatSetpoint:Get({setpoint_type = ThermostatSetpoint.setpoint_type.HEATING_1}))
+    device:set_field(SETPOINT_REPORT_QUEUE, function ()
+      device:send(ThermostatSetpoint:Get({setpoint_type = ThermostatSetpoint.setpoint_type.HEATING_1}))
+    end)
   else
     device:send(ThermostatSetpoint:Get({setpoint_type = ThermostatSetpoint.setpoint_type.HEATING_1}))
-    device:send(ThermostatSetpoint:Get({setpoint_type = ThermostatSetpoint.setpoint_type.COOLING_1}))
+    device:set_field(SETPOINT_REPORT_QUEUE, function ()
+      device:send(ThermostatSetpoint:Get({setpoint_type = ThermostatSetpoint.setpoint_type.COOLING_1}))
+    end)
```

Comment:
Is the intention for `SETPOINT_REPORT_QUEUE` to be overridden or appended to here and in the `set_setpoint_factory` closure? I think there is a case where the driver receives a setpoint set cap cmd and the multilevel get and operating state get commands are queued, but before sending the commands a mode report is received and the setpoint get command overrides the previous two commands so they are never sent.

Reply:
It can happen, but it's not a situation I'm too concerned about.

Reply:
Based on current behavior of the device that situation would already be an issue. I think as long as we're getting _any_ setpoint reports, it's an improvement.

Reply:
If its an improvement, it seems worthwhile.